graphics/src/MeshManager.cc

```diff
@@ -822,6 +822,87 @@ void MeshManager::CreateCamera(const std::string &_name, float _scale)
   mesh->RecalculateNormals();
 }
 
+void MeshManager::CreateEllipsoid(const std::string &_name,
+                                  const ignition::math::Vector3d _radii,
+                                  const unsigned int _rings,
+                                  const unsigned int _segments)
+{
+  if (this->HasMesh(_name))
+  {
+    return;
+  }
+
+  Mesh *mesh = new Mesh();
+  mesh->SetName(_name);
+  this->dataPtr->meshes.insert(std::make_pair(_name, mesh));
+
+  ignition::math::Vector3d vert, norm;
+  SubMesh subMesh;
+
+  double umin = -M_PI / 2.0;
+  double umax = M_PI / 2.0;
+  double vmin = 0.0;
+  double vmax = 2.0 * M_PI;
+
+  unsigned int i, j;
+  float theta, phi;
+  float d_phi = (umax - umin) / (_rings - 1.0);
+  float d_theta = (vmax - vmin) / (_segments - 1.0);
```

Comment:
Should these be `double` since everything else is?

```suggestion
  double theta, phi;
  const double d_phi = (umax - umin) / (_rings - 1.0);
  const double d_theta = (vmax - vmin) / (_segments - 1.0);
```

Reply:
Still seeing a mismatch here. I don't think the precision should matter too much, but it is better to be consistent. 

If we are using `float`, then `cos` and `sin` need to be changed to `cosf` and `sinf`, and `ignition::math::Vector3d` should be `Vector3f` I believe.

If we are using double, these should be marked as such.